pkg/assisted/agentserviceconfig.go

```diff
@@ -24,6 +25,10 @@ const (
 	defaultImageStoreStorageSize = "10Gi"
 )
 
+var (
+	validIPXEOptions = []string{"enabled", "disabled"}
+)
+
 // AgentServiceConfigBuilder provides struct for the agentserviceconfig object containing connection to
 // the cluster and the agentserviceconfig definition.
 type AgentServiceConfigBuilder struct {
```

Comment:
NewAgentServiceConfigBuilder returns nill if apiClient is empty.
It'll cause panic in case if we run multiple functions in one line. For example:
`NewAgentServiceConfigBuilder().With..().With..().Create()
`
Please refer to https://github.com/openshift-kni/eco-goinfra/pull/418/files to fix this issue.

Reply:
Same relevant for NewDefaultAgentServiceConfigBuilder

Reply:
I'm not sure I follow, the validate method should prevent the panic from happening:
Code:
```
	apiClient := clients.New("")

	_, err := assisted.NewDefaultAgentServiceConfigBuilder(apiClient).WithUnauthenticatedRegistry("blah").Create()
	if err != nil {
		fmt.Println(err)
	}
```
Output:
```
2024/05/23 10:06:31 Using in-cluster kube client config
error: received nil AgentServiceConfig builder
```